src/common/dump.c

```diff
@@ -143,42 +140,40 @@ int dump_save_to_usb(const char *fn) {
     uint8_t buff[DUMP_BUFF_SIZE];
     int bw;
     int bw_total = 0;
-    if (w25x_init()) {
-        fd = fopen(fn, "w");
-        if (fd != NULL) {
-            //save dumped RAM and CCRAM from xflash
-            for (addr = 0; addr < DUMP_XFLASH_SIZE; addr += DUMP_BUFF_SIZE) {
-                memset(buff, 0, DUMP_BUFF_SIZE);
-                w25x_rd_data(addr, buff, DUMP_BUFF_SIZE);
-                bw = fwrite(buff, 1, DUMP_BUFF_SIZE, fd);
-                if (bw <= 0) {
-                    break;
-                }
-                bw_total += bw;
-            }
-            //save OTP
-            for (addr = 0; addr < DUMP_OTP_SIZE; addr += DUMP_BUFF_SIZE) {
-                bw = fwrite((void *)(DUMP_OTP_ADDR + addr), 1, DUMP_BUFF_SIZE, fd);
-                if (bw <= 0) {
-                    break;
-                }
-                bw_total += bw;
+    fd = fopen(fn, "w");
+    if (fd != NULL) {
+        //save dumped RAM and CCRAM from xflash
+        for (addr = 0; addr < DUMP_XFLASH_SIZE; addr += DUMP_BUFF_SIZE) {
+            memset(buff, 0, DUMP_BUFF_SIZE);
+            w25x_rd_data(addr, buff, DUMP_BUFF_SIZE);
+            bw = fwrite(buff, 1, DUMP_BUFF_SIZE, fd);
+            if (bw <= 0) {
+                break;
```

Comment:
Open file leak

Reply:
I don't see it. Can you elaborate more? note: this is `break` statement and not `return` statement.

Reply:
my bad, I don't see it either! :)